src/Parsers/ExpressionListParsers.cpp

```diff
@@ -2333,7 +2333,7 @@ std::unique_ptr<Layer> getFunctionLayer(ASTPtr identifier, bool is_table_functio
 
     /// NOT is always parsed as an operator, because there is no difference between
     /// the operator form: NOT (1 + 2), and the functional form: not(1 + 2), when the operand has to be in parentheses.
-    if (function_name_lowercase == "not" && !is_table_function)
+    if (function_name_lowercase == "not" && (!is_table_function || !is_first_identifier))
```

Comment:
@alexey-milovidov do you think this would work?
There is a high rate of failures related to formatting of `not` so let's get one of the PRs merged.